src/vmm/src/rpc_interface.rs

```diff
@@ -468,6 +473,15 @@ impl RuntimeApiController {
             GetVmConfiguration => Ok(VmmData::MachineConfiguration(
                 self.vm_resources.vm_config().clone(),
             )),
+            GetVmInstanceInfo => Ok(VmmData::InstanceInformation(
+                // Clone is redundant only for test MockVmm.
+                #[allow(clippy::redundant_clone)]
```

Comment:
Woldn't it make sense to have `Vmm::instance_info` return an `InstanceInfo` instead of a reference and have the clone inside that function?

Reply:
sure, that would work! I tried to adhere to the general practice of having a reference getter, but indeed we could just pass a copy and be done with it.

Reply:
We took the same approach (passing a copy) with both the `GetBalloonStats` and `GetBalloonConfig` requests. The structs were small in size, so a `clone` didn't seem like a big deal at the time. I think it applies to this context as well.